security/advancedtls/crl.go

```diff
@@ -303,24 +303,30 @@ func fetchCRL(c *x509.Certificate, crlVerifyCrt []*x509.Certificate, cfg Revocat
 		if crl == nil {
 			return nil, fmt.Errorf("no CRL found for certificate's issuer")
 		}
+		if err := verifyCRL(crl, nil, crlVerifyCrt); err != nil {
+			return nil, fmt.Errorf("verifyCRL() failed: %v", err)
+		}
 		return crl, nil
 	}
 	return fetchIssuerCRL(c.RawIssuer, crlVerifyCrt, cfg)
 }
 
-// checkCert checks a single certificate against the CRL defined in the certificate.
-// It will fetch and verify the CRL(s) defined in the root directory specified by cfg.
-// If we can't load any authoritative CRL files, the status is RevocationUndetermined.
+// checkCert checks a single certificate against the CRL defined in the
+// certificate. It will fetch and verify the CRL(s) defined in the root
+// directory (or a CRLProvider) specified by cfg. If we can't load any valid
+// authoritative CRL files, the status is RevocationUndetermined.
 // c is the certificate to check.
 // crlVerifyCrt is the group of possible certificates to verify the crl.
 func checkCert(c *x509.Certificate, crlVerifyCrt []*x509.Certificate, cfg RevocationConfig) RevocationStatus {
 	crl, err := fetchCRL(c, crlVerifyCrt, cfg)
 	if err != nil {
-		// We couldn't load any CRL files for the certificate, so we don't know
-		// if it's RevocationUnrevoked or not.  This is not necessarily a
+		// We couldn't load any valid CRL files for the certificate, so we don't
+		// know if it's RevocationUnrevoked or not. This is not necessarily a
 		// problem - it's not invalid to have no CRLs if you don't have any
-		// revocations for an issuer. We just return RevocationUndetermined and
-		// there is a setting for the user to control the handling of that.
+		// revocations for an issuer. It also might be an indication of CRL file to
```

Comment:
"It also might be an indication of CRL file to be invalid" -> "It also might be an indication that the CRL file in invalid."

Reply:
Done